resources/active_directory.rb

```diff
@@ -0,0 +1,55 @@
+#
+# Cookbook:: sensu-go
+# Resource:: agent
+#
+# Copyright:: 2018 Sensu, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+include SensuCookbook::SensuMetadataProperties
+include SensuCookbook::SensuCommonProperties
+
+resource_name :sensu_active_directory
+
+action_class do
+  include SensuCookbook::Helpers
+end
+
+property :ad_servers, Array, required: true
+property :groups_prefix, String
+property :username_prefix, String
+property :resource_type, String, default: 'ad'
+alias_method :servers, :ad_servers
```

Comment:
I think this needs to be reversed or the property needs to be renamed as `servers` to alias it.

Reply:
I thought that might come up, so I added a test for both regular AD and the alias: https://travis-ci.org/github/sensu/sensu-go-chef/jobs/694743840#L892